sktime/distances/mpdist.py

```diff
@@ -0,0 +1,275 @@
+    element_wise_mult = np.multiply(t_fft, q_fft)
+    reverse_fft = np.fft.ifft(element_wise_mult)
+
+    # Returns only the valid dot products from reverse_fft
+    dot_prod = reverse_fft[q_len-1:t_len].real
+
+    return dot_prod
+
+def compute_mean_std(q, t, q_len, t_len, n_t_subs):
+    """
+
+        Computes the mean and standard deviation of the query and the time series.
+
+        Parameters
+        ----------
+            q: pandas.Series
+                Query.
+            t: pandas.Series
+                Time series.
+            q_len: int
+                Length of the query.
+            t_len: int
+                Length of the time series.
+            n_t_subs: int
+                Number of subsequences in the time series.
+
+        Output
+        ------
+            q_mean: float
+                    Mean of the elements from the query.
+            q_std: float
+                    Standard deviation of the elements from the query.
+            t_mean: numpy.array
+                    Array with the mean of the elements from each subsequence of length(query) from the time series.
+            t_std: numpy.array
+                    Array with the standard deviation of the elements from each subsequence of length(query) from the time series.
+
+    """
+
+    # Query
+    q_mean = np.mean(q)
+    q_std = np.std(q)
+
+    # Time series
+    t_mean = np.empty(n_t_subs)
+
+    for i in range(0, n_t_subs):
+        t_mean[i] = np.mean(t[i:i+q_len])
+
+    t_std = np.empty(n_t_subs)
+
+    for i in range(0, n_t_subs):
+        t_std[i] = np.std(t[i:i+q_len])
+
+    return q_mean, q_std, t_mean, t_std
+
+def calculate_distance_profile(q, t, dot_prod, q_mean, q_std, t_mean, t_std, q_len, t_len, n_t_subs):
+    """
+
+        Calculates the distance profile for the given query.
+
+        Parameters
+        ----------
+            q: pandas.Series
+                Query.
+            t: pandas.Series
+                Time series.
+            dot_prod: numpy.array
+                Sliding dot products between the time series and the query.
+            q_mean: float
+                Mean of the query's elements.
+            q_std: float
+                Standard deviation of query's elements.
+            t_mean: numpy.array
+                Array with the mean of the elements from each subsequence of length(query) from the time series.
+            t_std: numpy.array
+                Array with the standard deviation of the elements from each subsequence of length(query) from the time series.
+            q_len: int
+                Length of the query.
+            t_len: int
+                Length of the time series.
+            n_t_subs: int
+                Number of subsequences in the time series.
+
+        Output
+        ------
+            d: numpy.array
+                Distance profile of query q.
+
+    """
+
+    d = np.zeros(n_t_subs)
+
+    for i in range(0, n_t_subs):
```

Comment:
You can increase the performance of this loop by creating a numpy array using the function inside the loop, e.g. 
`d = np.array([2*q_len*(1-((dot_prod[i]-q_len*q_mean*t_mean[i])/(q_len*q_std*t_std[i])) for i in range(0, n_t_subs)])
d = np.absolute(d)
d = np.sqrt(d)`

^ that gets around the initialisation of d to zeros and native abs python function :)